cmd/main.go

```diff
@@ -121,19 +124,47 @@ func serverSetup(options *Options, logger *zap.Logger) {
 		logger.Fatal("Failed in initialize s3 COS driver", zap.Error(err))
 		os.Exit(1)
 	}
-	serveMetrics(options.MetricsAddress, logger)
+	serveMetrics(options.ServerMode, options.MetricsAddress, logger)
 	S3CSIDriver.Run()
 }
 
-func serveMetrics(metricsAddress string, logger *zap.Logger) {
-	logger.Info("starting metrics endpoint")
+func serveMetrics(mode, metricsAddress string, logger *zap.Logger) {
+	logger.Info("starting metrics & socket-health endpoints")
+	http.Handle("/metrics", promhttp.Handler())
+	if strings.Contains(mode, "node") {
+		http.HandleFunc("/socket-health", func(w http.ResponseWriter, _ *http.Request) {
+			if err := checkSocketHealth(); err != nil {
+				http.Error(w, "unhealthy: "+err.Error(), http.StatusInternalServerError)
+				return
+			}
+			w.WriteHeader(http.StatusOK)
+			_, _ = w.Write([]byte("ok"))
+		})
+	}
+
 	go func() {
-		http.Handle("/metrics", promhttp.Handler())
-		//http.Handle("/health-check", healthCheck)
-		err := http.ListenAndServe(metricsAddress, nil) // #nosec G114: use default timeout.
-		logger.Error("failed to start metrics service:", zap.Error(err))
+		if err := http.ListenAndServe(metricsAddress, nil); err != nil { // #nosec G114: use default timeout.
+			logger.Error("failed to start metrics & socket-health service:", zap.String("addr", metricsAddress), zap.Error(err))
```

Comment:
`failed to start metrics & socket-health service:` --> `failed to start metrics & cos-csi-mounter socket-health service:`